### DISCUSSION TITLE
Models not showing for 'users' but showing for 'admin'

### DISCUSSION THREAD

**Discussion by williamzebrowskI** [open-webui/open-webui#4468] [General]
I have searched for a setting that would allow 'users' the ability to see models.  If I change the users permissions to a 'admin' the models show up.  Is there a reason for this or a settings I'm not seeing?

**Comment by HritwikSinghal** [comment]
@williamzebrowskI Hey, just marking issues as solved without explanation is not super helpful. Can you please share what you did to fix this so I (and others) can learn from it?. 

I'll post my solution.

Go to "adimn Panel" -> "settings" -> "models" and edit each model's visibility to public. DO NOT forget to click "save & update".

**Comment by cschmittiey** [comment]
I have 50+ models in my list. Is it expected we have to edit every single one individually? Hoping someone comes across a way to do this programmatically.

**Comment by ezbarcelos** [comment]
Not straightforward as one would expect, but it's actually quite simple.  

Go to **Admin Settings** > Users  (top nav bar) and select **Groups**.  Then click on "Default permissions" and set _Models Access_ to **True**. Don't forget to hit the Save button. Voilá!

**Reply by CmdQ** [reply]
Looks like the most sensible solution. Unfortunately, it doesn't do it for me.

**Reply by peter-hartmann** [reply]
Admin Settings > Users > Groups > Default permissions enables the ```Model``` tab under ```Workspace```, but doesn't enable any models for ```Users``` for me.

**Reply by alexk-ede** [reply]
It tried that, too bc it looked sensible, but it's not working for me either.

**Reply by ihrke** [reply]
same here

**Reply by jugurtha114** [reply]
same here, it doesnt work

**Reply by tugbayatilla** [reply]
not working as expected, nope

**Reply by ezbarcelos** [reply]
Coming back here to clarify. You **will** get this to work....

Consider this use case --  

Suppose you have 2 users:  `user1@yourdomain.com` and `user2@yourdomain.com`
None are admin. **User 1** is part of a "My-Team" group. **User 2** is _not_ part of any group.
The "My-Team" group has "Model Access" set to ON under permissions.

Also, consider you are serving 4 models via Ollama. Within Open-WebUI's admin settings, you go and set _Model 1_ and _Model 2_ to **public** as per [HritwikSinghal](https://github.com/HritwikSinghal)'s instructions in this thread (ie. Admin Panel > Settings > Models - and change models 1 and 2's visibility to public. Hit "save & update").

🔑 **The key**: 
Still under Admin Panel > Settings > Models, now edit _Model 3_. 
Keep it **private**, but add the "My-Team" group under **Groups** and save.

No changes to Model 4.

All in all, if you had Models Access set to True under "Default permissions" as per my previous post, both users 1 and 2 would have visibility to models 1 and 2, while only User 1 (part of the "My-Team" group) would have access to Model 3.  

What about Model 4? Only _admins_ would see it.

💡These settings allow you to set a combination of "plans" (think free, pro, premium) and preferences (language, vision, etc).

Enjoy. 👍

**Comment by peter-hartmann** [comment]
```BYPASS_MODEL_ACCESS_CONTROL=True```
... setting this environment worked for me, it enables all models for all users.

Or:
Admin Panel > Settings > Models, and then edit each Model and change ```Visibility``` to ```Public``` to enable that model for ```User```.

**Comment by proggod** [comment]
Same with me, can't add models to groups or even go in model and select a group to show, only if you make public they show, so no way to assign specific models to people.

**Comment by stevenh** [comment]
Without a easier way to allocated access to models, pipe manifolds and connections which provide multiple models require a large amount of effort to setup.

Currently the only workaround is setting `BYPASS_MODEL_ACCESS_CONTROL=true` which doesn't provide any granular control.

Curious why this was closed, without a proper resolution?

**Comment by nonlinear** [comment]
the confusion is that openwebui calls "models" both for LLM and agents. 

I think OP means that users can't create (agent) models because they don't have access to (LLM) models.

there's a variable that changes it, `BYPASS_MODEL_ACCESS_CONTROL=true`

**Comment by palexisd** [comment]
To quickly modify the access of your models, you can export the config as .json, change the value `"access_control":` to `"null"` and re-import the config. 

The "null" value makes the models public.

**Reply by raghavdasila** [reply]
this isn't working with openrouter direct connection. it seems it doesn't update existing presets, but manual changes persist

**Comment by jt-datastation** [comment]
I found the setting in Admin > Models > [choose model] > [Top right lock icon with the word Access] > [Change from Private to Public] > Save.

**Comment by DenisMkS** [comment]
I have other way around issue. Model is shown for users, but I don't see it in admin panel. However, I can see it in "Reorder model"

**Reply by DenisMkS** [reply]
> I have other way around issue. Model is shown for users, but I don't see it in admin panel. However, I can see it in "Reorder model"

Found it. Since it was cloned from another model, it can be found in "Workspace"
